### ISSUE TITLE
Add TypeScript support

### ISSUE THREAD

**Issue by tkrotoff** [jharris4/html-webpack-tags-plugin#52]
Current type definition I use in my projects (something like `types/html-webpack-tags-plugin.d.ts`):

```TypeScript
declare module 'html-webpack-tags-plugin' {
  import { Plugin } from 'webpack';

  interface Options {
    append?: boolean;
    prependExternals?: boolean;
    jsExtensions?: string | string[];
    cssExtensions?: string | string[];
    useHash?: boolean;
    addHash?: (assetPath: string, hash: string) => string;
    hash?: boolean | Function;
    usePublicPath?: boolean;
    addPublicPath?: (assetPath: string, publicPath: string) => string;
    publicPath?: boolean | string | Function;

    tags?: string | object | Array<string | object>;
    links?: string | object | Array<string | object>;
    scripts?: string | object | Array<string | object>;
    metas?: string | object | Array<string | object>;
  }

  export default class HtmlWebpackTagsPlugin extends Plugin {
    constructor(options: Options);
  }
}
```

Then you have autocompletion using vscode or whatever smart editor/IDE:

```TypeScript
// webpack.config.ts
import HtmlWebpackTagsPlugin from 'html-webpack-tags-plugin';

...
new HtmlWebpackTagsPlugin({...});
...
```

3 solutions (see https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html):
- add the typings inside this repo
- contribute to https://github.com/DefinitelyTyped/DefinitelyTyped
- use TypeScript directly

**Comment by jharris4** [comment]
Can you elaborate on what this means? Are you just looking for TypeScript types?

If so, a PR would be welcome...

**Comment by tkrotoff** [comment]
> looking for TypeScript types?

Yes, see edited issue description.

**Comment by jharris4** [comment]
Ok, I’m new to TypeScript but I’m curious.

Thanks for sharing your starter definition, I’ll see if I can find time to get around to adding it next week.

**Comment by jharris4** [comment]
I've added the TypeScript definitions and released in `2.0.16`.

@tkrotoff Can you please try out the definitions in the new version and let me know if it works for you so I can close this issue? Thanks!

**Comment by tkrotoff** [comment]
File types.d.ts is not part of the package: https://unpkg.com/browse/html-webpack-tags-plugin@2.0.16/

I guess you have to add types.d.ts to `files` inside package.json: https://docs.npmjs.com/files/package.json#files
```JSON
  ...
  "files": [
    "index.js",
    "types.d.ts"
  ],
  ...
```

**Comment by jharris4** [comment]
Oh crap! yeah I only tested locally with `import HtmlWebpackTagsPlugin from "../typings.d.ts";`

I'll publish another version with the file added. Sorry about that, and thanks for reporting it!

**Comment by jharris4** [comment]
Published in `2.0.17`. That should do the trick.

**Comment by tkrotoff** [comment]
It works fine, thx!
